File: 0547-number-of-provinces/0547-number-of-provinces.cpp
class Solution {
public:
  
void dfs( int u , vector<vector<int>>&graph , vector<int>&vis )
{
    vis[u]=1;
    for( int i=0 ;i< graph[u].size() ;i++ )
    {
        if( !vis[graph[u][i]] )
        {
            dfs(graph[u][i] , graph ,vis);
        }
    }
}


int findCircleNum(vector<vector<int>>& isConnected)
{
    vector<vector<int>>graph(isConnected.size());
    for( int i=0 ;i< isConnected.size();i++)
    {
        for(int j=i ;j<isConnected[i].size() ;j++ )
        {
            if(isConnected[i][j])
            {
                graph[i].push_back(j);
                graph[j].push_back(i);
            }
        }
    }
    int n =0;
    vector<int>vis(isConnected.size() ,0);
    for(int i=0 ;i< isConnected.size() ;i++ )
    {
        if( !vis[i])
        {
            dfs(i,graph,vis);
            n++;
        }
    }
    return n;

}
};

File: 0547-number-of-provinces/0547-number-of-provinces.js
/**
 * @param {number[][]} isConnected
 * @return {number}
 */
var dfs = function (node, visited, isConnected) {
  visited[node] = 1;
  for (let i = 0; i < isConnected.length; i++) {
    if (isConnected[node][i] && !visited[i]) {
      dfs(i, visited, isConnected);
    }
  }
};

var findCircleNum = function (isConnected) {
  var visited = new Array(isConnected.length).fill(0);

  // for (let i = 0; i < isConnected.length; i++) {
  //   visited[i] = new Array(isConnected.length).fill(0);
  // }

  var count = 0;
  for (let node = 0; node < isConnected.length; node++) {
    if (!visited[node]) {
      dfs(node, visited, isConnected);
      count++;
    }
  }

  console.log(visited);
  return count;
};